### PR TITLE
[Enhancement/Item] Add Achievement to Unlock Eviolite

### PR DESCRIPTION
**Pull request by xsn34kzx**
## What are the changes?

Adds a new achievement and unlock condition that, upon satisfying its requirement (completing Classic Mode with at least one Pokemon that can still evolve), introduces `Eviolite` into the item pool for future runs à la `Mini Black Hole`. For players, this would effectively be the official introduction of `Eviolite` as an item.

## Why am I doing these changes?

When `Eviolite` was put into the game (#2161), it was purposefully left out of the general item pool to be obtainable via other means. This PR follows up by actually introducing `Eviolite` as an unlockable item, which would address the ongoing demand for the item mentioned in its original PR.

## What did change?

- `EVIOLITE` was added to the `Unlockables` enum along with a corresponding case in `getUnlockableName` (as a fix, `Mini Black Hole` was adjusted to use its translated name in this function since it was hard-coded to English only previously). 
- The achievement, `Bring Your Child to Work Day`, was added such that its completion is validated on run clear.
- `Eviolite`'s entry in the item pool was brought back with its original weight function wrapped in an `if` checking if the item was unlocked.

#### Achievement Notification

achievementGet

#### Achievement Menu after Completion

achievementScreen

### Notes

- ~~With this achievement, the Achievement screen officially requires a scroll bar or icon size adjustment. I'm not well versed in UI-related changes, so this is something I'd like someone else to address either within this PR or another.~~
  - https://github.com/pagefaultgames/pokerogue/pull/3395
- On fresh data, everything fills in correctly; however, because this is a new unlock, it's relatively unclear if it'll be saved correctly since `unlocks` in `GameData` won't have an entry initializing the value for `EVIOLITE` to `false` on older saves initially. I couldn't test this situation directly after clearing my own saves, so feedback on existing cache data or in `beta` itself will be needed to make sure.

## How to test the changes?

An easy way to test whether the achievement works is by using the following override and picking any starter for a Classic Mode run:
```ts
STARTING_WAVE_OVERRIDE: 201
```
This should immediately cause Classic Mode to complete and the achievement to appear, and the text during the "Congratulations!" screen should let you know that `Eviolite` has been unlocked.

The following override can then be used in a followup run to re-roll rewards until the item appears:
```ts
WAIVE_ROLL_FEE_OVERRIDE: true
```

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [ ] Have I considered writing automated tests for the issue?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/locales/de/achv.ts [hunk 1]
```diff
@@ -170,6 +170,11 @@ export const PGMachv: AchievementTranslationEntries = {
     name: "Ungeschlagen",
     description: "Beende den klassischen Modus erfolgreich.",
   },
+  "UNEVOLVED_CLASSIC_VICTORY": {
+    name: "Bring Your Child To Work Day",
+    description: "Beat the game in Classic Mode with at least one unevolved party member."
```
Comment (CodeTappert) [inline src/locales/de/achv.ts:175]:
```suggestion
    description: "Beende den klassischen Modus erfolgreich mit mindestens einem nicht entwickeltem Pokémon in deinem Team"
```

File: src/locales/de/achv.ts [hunk 2]
```diff
@@ -170,6 +170,11 @@ export const PGMachv: AchievementTranslationEntries = {
     name: "Ungeschlagen",
     description: "Beende den klassischen Modus erfolgreich.",
   },
+  "UNEVOLVED_CLASSIC_VICTORY": {
+    name: "Bring Your Child To Work Day",
```
Comment (CodeTappert) [inline src/locales/de/achv.ts:174]:
```suggestion
    name: "'Bringe dein Kind zur Arbeit'-Tag" ,
```

**Review by damocleas — APPROVED** [review]

**Review by Tempo-anon — CHANGES_REQUESTED** [review]
Gonna place a blocker on this for now until we get scrolling in the achievement screen

File: src/locales/ko/achv.ts [hunk 3]
```diff
@@ -169,6 +169,10 @@ export const PGMachv: AchievementTranslationEntries = {
     name: "무패",
     description: "클래식 모드 클리어",
   },
+  "UNEVOLVED_CLASSIC_VICTORY": {
+    name: "Bring Your Child To Work Day",
+    description: "Beat the game in Classic Mode with at least one unevolved party member."
+  },
```
Comment (returntoice) [inline src/locales/ko/achv.ts:175]:
```suggestion
    name: "우리집 꿈나무",
    description: "최종 진화형이 아닌 포켓몬을 데리고 클래식 모드 클리어."
  },
```

**Comment by EnochG1** [comment]
@xsn34kzx 
From now Japanese is added in beta.
Could you merge latest beta and add Japanese locale files? 🙏

**Review by CodeTappert — APPROVED** [review]

File: src/locales/fr/achv.ts [hunk 4]
```diff
@@ -169,6 +169,10 @@ export const PGMachv: AchievementTranslationEntries = {
     name: "Invaincu",
     description: "Terminer le jeu en mode classique",
   },
+  "UNEVOLVED_CLASSIC_VICTORY": {
+    name: "Bring Your Child To Work Day",
+    description: "Beat the game in Classic Mode with at least one unevolved party member."
```
Comment (Adri1) [inline src/locales/fr/achv.ts:174]:
```suggestion
    name: "Stagiaire de 3e",
    description: "Terminer le mode Classique avec au moins un Pokémon non-évolué dans l’équipe."
```

**Review by Adri1 — COMMENTED** [review]
Added French, but could you also add it to the female achivments?

File: src/locales/pt_BR/achv.ts [hunk 5]
```diff
@@ -169,6 +169,10 @@ export const PGMachv: AchievementTranslationEntries = {
     name: "Invencível",
     description: "Vença o jogo no modo clássico",
   },
+  "UNEVOLVED_CLASSIC_VICTORY": {
+    name: "Bring Your Child To Work Day",
+    description: "Beat the game in Classic Mode with at least one unevolved party member."
+  },
```
Comment (josericardo-fo) [inline src/locales/pt_BR/achv.ts:175]:
```suggestion
  "UNEVOLVED_CLASSIC_VICTORY": {
    name: "Tire as Crianças da Sala",
    description: "Vença o jogo no Modo Clássico com pelo menos um membro da equipe não evoluído.."
  },
```

File: src/locales/fr/achv.ts [hunk 6]
```diff
@@ -439,6 +443,10 @@ export const PGFachv: AchievementTranslationEntries = {
     name: "Invaincue",
     description: "Terminer le jeu en mode classique",
   },
+  "UNEVOLVED_CLASSIC_VICTORY": {
+    name: "Bring Your Child To Work Day",
+    description: "Beat the game in Classic Mode with at least one unevolved party member."
```
Comment (Adri1) [inline src/locales/fr/achv.ts:448]:
```suggestion
    name: "Le stagiaire de 3e",
    description: "Terminer le mode Classique avec au moins un Pokémon non-évolué dans l’équipe."
```

**Pull request merged**